scalardl/src/scalardl/core.clj

```diff
@@ -68,6 +69,25 @@
       (prepare-client-service test))
     client-service))
 
+(defn register-certificate
+  [client-service]
+  (let [resp (.registerCertificate client-service)]
+    (when-not (util/success? resp)
+      (throw (ex-info "Failed to register a certificate"
+                      {:cause "Failed to register a certificate"})))))
+
+(defn register-contracts
+  "Register contracts which have
+  {:name contract-name, :class class-name, :path contract-path}"
+  [client-service contracts]
+  (doseq [c contracts]
+    (let [resp (.registerContract client-service
+                                  (:name c) (:class c) (:path c)
+                                  (Optional/empty))]
+      (when-not (util/success? resp)
+        (throw (ex-info "Failed to register a contract"
```

Comment:
Just a question: is it worth also including the contract name or perhaps path here? It may make it easier to find the cause of the error.

Reply:
Thanks! I've added contract info. in the exception.